ebean-core/src/main/java/io/ebeaninternal/server/persist/DefaultPersister.java

```diff
@@ -642,146 +642,222 @@ public int delete(Class<?> beanType, Object id, Transaction transaction, boolean
 
     id = descriptor.convertId(id);
     DeleteMode deleteMode = (permanent || !descriptor.isSoftDelete()) ? DeleteMode.HARD : DeleteMode.SOFT;
-    return delete(descriptor, id, null, transaction, deleteMode);
+    return delete(descriptor, id, transaction, deleteMode);
   }
 
   @Override
   public int deleteByIds(BeanDescriptor<?> descriptor, List<Object> idList, Transaction transaction, boolean permanent) {
     DeleteMode deleteMode = (permanent || !descriptor.isSoftDelete()) ? DeleteMode.HARD : DeleteMode.SOFT;
-    return delete(descriptor, null, idList, transaction, deleteMode);
+    return delete(descriptor, idList, transaction, deleteMode);
   }
 
-  /**
-   * Delete by Id or a List of Id's.
-   */
-  private int delete(BeanDescriptor<?> descriptor, Object id, List<Object> idList, Transaction transaction, DeleteMode deleteMode) {
+  private int delete(BeanDescriptor<?> descriptor, List<Object> idList, Transaction transaction, DeleteMode deleteMode) {
     if (idList == null || idList.size() <= maxDeleteBatch) {
```

Comment:
pretty sure idList is never null here now, but we want to adjust this part anyway using number of id properties